### PR TITLE
Add global fee settings from config.js

### PR DESCRIPTION
**Pull request by lndhub-admin**
This allows for setting fees in config.js that will be read and set at startup.

The changes also allows for dynamic fee settings at runtime through Extension Admin API,
demonstrated in a follow up PR to this one will be a example UI admin extension.

### CODE REVIEW

File: class/Paym.js [hunk 1]
```diff
@@ -12,9 +12,9 @@ export class Paym {
     this._isPaid = null;
   }
 
-  static get fee() {
-    return 0.003;
-  }
+  // static get fee() {
+  //   return 0.003;
+  // }
```
Comment (summatix) [inline class/Paym.js]: Should remove code instead of commenting it out

File: config.js [hunk 2]
```diff
@@ -2,6 +2,8 @@ let config = {
   enableUpdateDescribeGraph: false,
   postRateLimit: 100,
   rateLimit: 200,
+  defaultForwardReserveFee: 0.01,
+  defaultIntraHubFee: 0.003,
```
Comment (summatix) [inline config.js]: The naming here seems inconsistent. `forwardReserveFee` and `intraHubFee` I think would be appropriate without the `default`.

File: controllers/api.js [hunk 3]
```diff
@@ -15,6 +15,12 @@ redis.monitor(function (err, monitor) {
   });
 });
 
+/****** START SET FEES FROM CONFIG AT STARTUP ******/
+/** GLOBALS */
+global.forwardFee = config.defaultForwardReserveFee;
+global.internalFee = config.defaultIntraHubFee;
```
Comment (summatix) [inline controllers/api.js]: Is there a way to pass these settings into Paym instead of relying on globals?

File: controllers/api.js [hunk 4]
```diff
@@ -258,8 +264,8 @@ router.post('/payinvoice', async function (req, res) {
         await u.savePaidLndInvoice({
           timestamp: parseInt(+new Date() / 1000),
           type: 'paid_invoice',
-          value: +info.num_satoshis + Math.floor(info.num_satoshis * Paym.fee),
-          fee: Math.floor(info.num_satoshis * Paym.fee),
+          value: +info.num_satoshis + Math.floor(info.num_satoshis * internalFee),
```
Comment (summatix) [inline controllers/api.js:267]: You could probably rely on the original `Paym.fee` property instead of creating a new global variable, and update what `Paym.fee` returns.

File: controllers/api.js [hunk 5]
```diff
@@ -312,7 +318,7 @@ router.post('/payinvoice', async function (req, res) {
       let inv = {
         payment_request: req.body.invoice,
         amt: info.num_satoshis, // amt is used only for 'tip' invoices
-        fee_limit: { fixed: Math.floor(info.num_satoshis * 0.005) + 1 },
+        fee_limit: { fixed: Math.floor(info.num_satoshis * internalFee) + 1 }, // fee setting was 0.005 now set as internalFee
```
Comment (summatix) [inline controllers/api.js]: `internalFee` is now set to `0.003` instead of `0.005`. Is this intentional?

File: config.js (hunk 2 above)
Comment (xraid) [reply config.js]: the variables at config represents default values at invocation that can later be adjusted at runtime, it also represents what was hard-coded as defaults, therefore chosen for clarity, i do not mind if You call them Bill & Linda ...

File: controllers/api.js (hunk 3 above)
Comment (xraid) [reply controllers/api.js]: Please try and report back ...

File: controllers/api.js (hunk 4 above)
Comment (xraid) [reply controllers/api.js:267]: Please try and report back ...

File: controllers/api.js (hunk 5 above)
Comment (xraid) [reply controllers/api.js]: I do belive this is a typo that has been lingering ? makes no sense have Paym.fee at 0.03 and then specify a *tip* fee at 0.05 ?

File: class/Paym.js (hunk 1 above)
Comment (xraid) [reply class/Paym.js]: Please do ...

File: class/User.js [hunk 6]
```diff
@@ -579,7 +579,7 @@ export class User {
       try {
         json = JSON.parse(paym);
         result.push(json);
-      } catch (_) {}
+      } catch (_) { }
```
Comment (AaronDewes) [inline class/User.js]: Please leave unrelated format changes out of the PR, and lint your code, because I think the linter didn't do it this way

**Review by 0racl3z — APPROVED** [review]
Thank you all soo much! Sry for the late reply, this has been quite the journey of finding a solution. (Which at this point i still haven't) I'll be sure to keep better tabs on projects!

File: class/User.js (hunk 6 above)
Comment (0racl3z) [reply class/User.js]: Thanks!

**Review by 0racl3z — APPROVED** [review]

**Comment by AaronDewes** [comment]
grafik

Really?

File: controllers/api.js [hunk 7]
```diff
@@ -15,6 +15,12 @@ redis.monitor(function (err, monitor) {
   });
 });
 
+/****** START SET FEES FROM CONFIG AT STARTUP ******/
+/** GLOBALS */
+global.forwardFee = config.defaultForwardReserveFee;
```
Comment (Overtorment) [inline controllers/api.js]: ok, if you intend to change them at runtime it makes sense to have them as global, instead of using `config.bla`.
but how often is this gonna happen? if you have some plugin code that edit those values, why not save them to config and request a restart of lndhub?

Comment (lndhub-admin) [reply controllers/api.js]: yes . true . this was 2 fold, both fixing the asked for fee settings functionality and at same time be able demonstrate LndHub-Admin extension example, where also login and then set new fees if wanted / needed can be had. It coincided and makes for a great demo, it was the quickest at hand for demo using the new feeSetting.

i find it handy be able set fees at runtime YMMV

Comment (lndhub-admin) [reply controllers/api.js]: also the umbrel ++ ppl need a UI setting fees can not easily edit dockerimages pulled from net and proficiency in commandline might not have developed for them yet !

Comment (lndhub-admin) [reply controllers/api.js]: but You say UI should write new config and then manually restart ?

can be problematic with different distros of node environments with paths and permissions and what not ?

**Review by 0racl3z — COMMENTED** [review]
wondering if the paym/class is an issue?

**Comment by lndhub-admin** [comment]
no issues all good, erlier before this patch the thing is Paym.js is a class and referenced a static method from in user.js and api.js and problem was having reach a regular variable when Paym.js instanciated without having to refactor much more then we try do now.

so all good and works . You can try it with UI checking out branch Admin-Extension-router . build and run ...
https://github.com/lndhub-admin/LndHub/tree/Admin-Extension-router

**Review by 0racl3z — APPROVED** [review]

**Review by 0racl3z — COMMENTED** [review]
thanks! i was wondering about the check that didn't pass?

**Comment by lndhub-admin** [comment]
OK

so i added 2 lines in config.js and 2 lines in controllers/api.js + 1 line in index.js = 5 lines

that enables fee settings at startup from config.js and also opens up for dynamic fee settings at runtime.

i also reduced LOC by 3 by deleting 3 lines in Paym.js

total add LOC 2, i myself regard that as rather elegant for the amount added functionality while not refactor wildly.

to be able continue https://github.com/lndhub-admin/LndHub-Admin-Extension i find the wait for PR's here are very slow and hinders more of PR's i would like to suggest namely 

- build and run with node instead of babel-node
- add to req.id middleware where reroutes for URL defined in config.js array be made possible

this all means LndHub core can be frozen while enabling being highly extensible

are you asking me to hard fork ? 
or is it that  You se any problems with this PR ?

thanks

**Review by 0racl3z — APPROVED** [review]
understood, thank you!

File: controllers/api.js [hunk 8]
```diff
@@ -312,7 +318,7 @@ router.post('/payinvoice', async function (req, res) {
       let inv = {
         payment_request: req.body.invoice,
         amt: info.num_satoshis, // amt is used only for 'tip' invoices
-        fee_limit: { fixed: Math.floor(info.num_satoshis * 0.005) + 1 },
```
Comment (Overtorment) [inline controllers/api.js:315]: feels like `0.005`  is a bug and actually `0.01` was intended here. thoughts?

**Review by Overtorment — APPROVED** [review]
lgtm.
this is more elegant then dancing around redis

File: controllers/api.js (hunk 8 above)
Comment (lndhub-admin) [reply controllers/api.js:315]: its seldom hit ? (a amt less invoice ?) anyway forwardFee in its place works for me altho now is set in this PR as internalFee

**Review by 0racl3z — APPROVED** [review]
thanks!
